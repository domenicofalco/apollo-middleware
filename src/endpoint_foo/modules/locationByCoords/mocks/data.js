export const locationByCoords = {
  id: "0000",
  value: "Torino"
};
