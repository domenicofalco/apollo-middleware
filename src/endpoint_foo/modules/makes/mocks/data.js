export const makes = {
  top: [
    {
      id: "id_audi",
      label: "AUDI",
      slugLabel: "audi"
    },
    {
      id: "id_bmw",
      label: "BMW",
      slugLabel: "bmw"
    }
  ],
  all: [
    {
      id: "id_abarth",
      label: "ABARTH",
      slugLabel: "abarth"
    }
  ]
};
