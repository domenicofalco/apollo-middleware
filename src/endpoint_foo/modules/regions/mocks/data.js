export const regions = [
  {
    id: "some_id_camp",
    value: "Campania"
  },
  {
    id: "some_id_lomb",
    value: "Lombardia"
  }
];
