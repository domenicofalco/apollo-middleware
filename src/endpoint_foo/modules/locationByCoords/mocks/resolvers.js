import { locationByCoords } from "./data";

export default {
  Query: () => ({
    locationByCoords: () => {
      return locationByCoords;
    }
  })
};
