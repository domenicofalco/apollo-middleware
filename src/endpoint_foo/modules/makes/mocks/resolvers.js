import { makes } from "./data";

export default {
  Query: () => ({
    makes: () => {
      return makes;
    }
  })
};
