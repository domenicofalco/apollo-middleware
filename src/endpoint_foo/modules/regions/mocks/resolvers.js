import { regions } from "./data";

export default {
  Query: () => ({
    regions: () => {
      return regions;
    }
  })
};
