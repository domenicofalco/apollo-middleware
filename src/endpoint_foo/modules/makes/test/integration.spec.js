// https://github.com/apollographql/fullstack-tutorial/blob/master/final/server/src/__tests__/integration.js
import { ApolloServer } from "apollo-server-express";
import { createTestClient } from "apollo-server-testing";
import { gql } from "apollo-server-express";
import { schema, dataSource } from "../";
import { makes } from "../mocks/data";

const { makesAPI } = dataSource;

const GET_MAKES = gql`
  query {
    makes(category: "cars") {
      top {
        id
        slugLabel
        label
      }
      all {
        id
        label
        slugLabel
      }
    }
  }
`;

describe("Makes", () => {
  it("fetches list of makes", async () => {
    const server = new ApolloServer({
      schema,
      dataSources: () => ({ makesAPI }),
      context: {}
    });

    makesAPI.getMakes = jest.fn(() => makes);

    const { query } = createTestClient(server);
    const res = await query({ query: GET_MAKES });

    expect(res).toMatchSnapshot();
  });
});
