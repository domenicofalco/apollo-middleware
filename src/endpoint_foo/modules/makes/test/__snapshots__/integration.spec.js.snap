// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Makes fetches list of makes 1`] = `
Object {
  "data": Object {
    "makes": Object {
      "all": Array [
        Object {
          "id": "id_abarth",
          "label": "ABARTH",
          "slugLabel": "abarth",
        },
      ],
      "top": Array [
        Object {
          "id": "id_audi",
          "label": "AUDI",
          "slugLabel": "audi",
        },
        Object {
          "id": "id_bmw",
          "label": "BMW",
          "slugLabel": "bmw",
        },
      ],
    },
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;
